pkg/client/messaging/client.go

```diff
@@ -69,6 +73,14 @@ type sendMsgOpts struct {
 	// Destination is service endpoint destination.
 	// This param can be used to send messages outside connection.
 	destination *service.Destination
+
+	// Message type of the response for the message sent.
+	// If provided then messenger will wait for the response of this type after sending message.
+	responseMsgType string
+
+	// context for await reply operation.
+	ctx    context.Context
```

Comment:
This isn't a generic ctx - it is tied to your waitForResponse. Should name appropriately.

Reply:
done